File: Part1_강의자료/Ch04_전체탐색_시뮬레이션/문제별코드/2745_진법변환/2745.cpp
#include <stdio.h>

char N[32];
int B;

int main()
{
    scanf("%s %d", N, &B);

    int num_dec = 0;
    for (int i = 0; N[i]; i++)
    {
        char digit = N[i];
        num_dec *= B;
        if ('0' <= digit && digit <= '9')
            num_dec += digit - '0';
        else
            num_dec += 10 + digit - 'A';
    }
    
    printf("%d\n", num_dec);
    return 0;
}

File: Part1_강의자료/Ch04_전체탐색_시뮬레이션/문제별코드/2745_진법변환/2745.py
# N, B = input().split()
# B = int(B)
# print(int(N, B))

N, B = input().split()
B = int(B)

num_dec = 0
for digit in N:
    num_dec *= B
    if '0' <= digit <= '9':
        num_dec += int(digit)
    else:
        num_dec += 10 + ord(digit) - ord('A')

print(num_dec)
